nixos/modules/services/networking/prosody.nix

```diff
@@ -1,887 +1,675 @@
-    };
+      };
 
-    smacks = mkOption {
-      type = types.bool;
-      default = true;
-      description = lib.mdDoc "Allow a client to resume a disconnected session, and prevent message loss";
-    };
+      config.settings = mkMerge [
+        (mkIf (config.module == "http_upload") {
+          http_upload_path = mkIf (config.module == "http_upload_path") cfg.dataDir;
+        })
+        (mkIf (config.module == "muc") {
+          modules_enabled = [ "muc_mam" ];
+        })
+      ];
+    }));
+    default = { };
+    description = mdDoc ''
+      Components are extra services on a server which are available to clients, usually on a subdomain
+      of the main server (such as mycomponent.example.com). Example components might be chatroom
+      servers, user directories, or gateways to other protocols.
+
+      See <https://prosody.im/doc/components> for details.
+    '';
+  };
 
-    # Admin interfaces
-    admin_adhoc = mkOption {
-      type = types.bool;
-      default = true;
-      description = lib.mdDoc "Allows administration via an XMPP client that supports ad-hoc commands";
-    };
+  acmeHosts = unique (mapAttrsToList (domain: hostOpts: hostOpts.useACMEHost) (filterAttrs (_: v: v.useACMEHost != null) cfg.virtualHosts));
+in
+{
+  imports = [
+    (mkRenamedOptionModule [ "services" "prosody" "allowRegistration" ] [ "services" "prosody" "settings" "allow_registration" ])
+    (mkRenamedOptionModule [ "services" "prosody" "httpPorts" ] [ "services" "prosody" "settings" "http_ports" ])
+    (mkRenamedOptionModule [ "services" "prosody" "httpInterfaces" ] [ "services" "prosody" "settings" "http_interfaces" ])
+    (mkRenamedOptionModule [ "services" "prosody" "httpsPorts" ] [ "services" "prosody" "settings" "https_ports" ])
+    (mkRenamedOptionModule [ "services" "prosody" "httpsInterfaces" ] [ "services" "prosody" "settings" "https_interfaces" ])
+    (mkRenamedOptionModule [ "services" "prosody" "c2sRequireEncryption" ] [ "services" "prosody" "settings" "c2s_require_encryption" ])
+    (mkRenamedOptionModule [ "services" "prosody" "s2sRequireEncryption" ] [ "services" "prosody" "settings" "s2s_require_encryption" ])
+    (mkRenamedOptionModule [ "services" "prosody" "s2sSecureAuth" ] [ "services" "prosody" "settings" "s2s_secure_auth" ])
+    (mkRenamedOptionModule [ "services" "prosody" "s2sInsecureDomains" ] [ "services" "prosody" "settings" "s2s_insecure_domains" ])
+    (mkRenamedOptionModule [ "services" "prosody" "s2sSecureDomains" ] [ "services" "prosody" "settings" "s2s_secure_domains" ])
+    (mkRenamedOptionModule [ "services" "prosody" "extraModules" ] [ "services" "prosody" "settings" "modules_enabled" ])
+    (mkRenamedOptionModule [ "services" "prosody" "extraPluginPaths" ] [ "services" "prosody" "settings" "plugin_paths" ])
+    (mkRenamedOptionModule [ "services" "prosody" "ssl" "key" ] [ "services" "prosody" "settings" "ssl" "key" ])
```

Comment:
This is remapped a bit oddly:
```
-ssl = {
-  cafile = "/etc/ssl/certs/ca-bundle.crt";
-  key = "/var/lib/acme/xxx/key.pem";
-  certificate = "/var/lib/acme/xxx/fullchain.pem";
-  
-};
+ssl = { ["cafile"] = "/etc/ssl/certs/ca-bundle.crt", ["certificate"] = "/var/lib/acme/xxx/fullchain.pem", ["key"] = "/var/lib/acme/xxx/key.pem" }
```
Indentation doesn't matter but I'm not sure prosody will consider `cafile = ...` the same as `["cafile"] = ...` ?

Reply:
My assumption (that I Believe I checked months ago but should check again) is that prosody uses `lua` to parse most of this config file so the square brackets are valid. I'll get back to you on this....

Reply:
after reading source pretty sure this is fine - i feel like my `prosody` server wouldn't have worked if it wasn't correct, especially given i can communicate with users on other jabber servers